File: src/VideoPublisher.java
import javafx.scene.control.TextField;
import javafx.scene.layout.HBox;
import javafx.scene.control.Label; // Añadimos Label para mostrar el nombre del publicador

public class VideoPublisher extends Publisher {
    private HBox view;

    public VideoPublisher(String name, Broker broker, String topic) {
        super(name, broker, topic); // Llama al constructor de Publisher

        Label nameLabel = new Label(name + " (" + topic + "): "); // Muestra el nombre del publicador y su tópico
        TextField urlField = new TextField();
        urlField.setPromptText("Ingrese URL del video");
        urlField.setPrefWidth(250); // Ajusta el ancho del campo de texto

        urlField.setOnAction(e -> {
            String url = urlField.getText();
            if (!url.isEmpty()) {
                // Publica la URL del video usando el método del Publisher
                publishNewEvent(url);
                urlField.clear(); // Limpia el campo de texto después de publicar
            }
        });

        view = new HBox(10);
        view.getChildren().addAll(nameLabel, urlField);
    }

    public HBox getView() {
        return view;
    }
}

File: src/VideoPublisher.cpp
#include "VideoPublisher.h"
#include <QHBoxLayout>

VideoPublisher::VideoPublisher(const std::string& name, Broker& broker, const std::string& topic, QWidget *parent)
        : Component(name, topic),
          QWidget(parent),
          Publisher(name, broker, topic) {

    // Crear la interfaz de usuario
    auto* layout = new QHBoxLayout(this);
    nameLabel = new QLabel(QString::fromStdString(name + " (" + topic + "): "), this);
    urlField = new QLineEdit(this);
    urlField->setPlaceholderText("Ingrese URL del video y presione Enter");

    layout->addWidget(nameLabel);
    layout->addWidget(urlField);
    setLayout(layout);

    // Conecta la señal 'returnPressed' (con Enter) del QLineEdit a nuestro slot de onUrlEntered
    connect(urlField, &QLineEdit::returnPressed, this, &VideoPublisher::onUrlEntered);
}

void VideoPublisher::onUrlEntered() {
    std::string url = urlField->text().toStdString();
    if (!url.empty()) {
        publishNewEvent(url); // Publica el evento
        urlField->clear(); // Limpia el campo de texto
    }
}